### PR TITLE
ARROW-6580: [Java] Support comparison for unsigned integers

### PR DESCRIPTION
**Pull request by liyafan82**
In this issue, we support the comparison of unsigned integer vectors, including UInt1Vector, UInt2Vector, UInt4Vector, and UInt8Vector.
With support for comparison for these vectors, the sort for them is also supported automatically.

### CODE REVIEW

**Comment by wesm** [comment]
@emkornfield @pravindra any thoughts on this?

File: java/algorithm/src/main/java/org/apache/arrow/algorithm/sort/DefaultVectorComparators.java [hunk 1]
```diff
@@ -56,6 +61,14 @@
...
       } else if (vector instanceof Float8Vector) {
         return (VectorValueComparator<T>) new Float8Comparator();
+      } else if (vector instanceof UInt1Vector) {
+        return (VectorValueComparator<T>) new UInt1Comparator();
+      } else if (vector instanceof UInt2Vector) {
+        return (VectorValueComparator<T>) new UInt2Comparator();
+      } else if (vector instanceof UInt4Vector) {
+        return (VectorValueComparator<T>) new UInt4Comparator();
+      } else if (vector instanceof UInt8Vector) {
+        return (VectorValueComparator<T>) new UInt8Comparator();
```
Comment (praveenbingo) [inline java/algorithm/src/main/java/org/apache/arrow/algorithm/sort/DefaultVectorComparators.java]: easier to read/more performant if we have map of class names to comparators?

File: java/algorithm/src/test/java/org/apache/arrow/algorithm/sort/TestDefaultVectorComparator.java [hunk 2]
```diff
@@ -107,4 +111,120 @@ public void testCompareLists() {
       assertTrue(comparator.compare(0, 0) == 0);
     }
   }
+
+  @Test
+  public void testCompareUInt1() {
```
Comment (praveenbingo) [inline java/algorithm/src/test/java/org/apache/arrow/algorithm/sort/TestDefaultVectorComparator.java]: can we please also add some test cases around boundary conditions (like Byte.Max and Byte.Min, Short.Max..)

File: java/algorithm/src/main/java/org/apache/arrow/algorithm/sort/DefaultVectorComparators.java (hunk 1 above)
Comment (liyafan82) [reply java/algorithm/src/main/java/org/apache/arrow/algorithm/sort/DefaultVectorComparators.java]: @praveenbingo Thanks a lot for your comments.
I agree with you that using a map with vector class as key is more performant. 

The reason that we use the vector to create the comparator is that, for some cases, the comparator cannot be created without the full knowledge of the vector.
For example, for a ListVector, we cannot create a comparator for it, unless we have information about its inner vector.

File: java/algorithm/src/test/java/org/apache/arrow/algorithm/sort/TestDefaultVectorComparator.java (hunk 2 above)
Comment (liyafan82) [reply java/algorithm/src/test/java/org/apache/arrow/algorithm/sort/TestDefaultVectorComparator.java]: Good suggestion. Thank you,
I have added test cases for the boundary conditions. Please take a look.

**Review by praveenbingo — APPROVED** [review]
lgtm

**Pull request closed without merging**